src/libs/actions/OnyxUpdateManager/index.ts

```diff
@@ -111,22 +108,22 @@ function handleOnyxUpdateGap(onyxUpdatesFromServer: OnyxEntry<OnyxUpdatesFromSer
     // The flow below is setting the promise to a reconnect app to address flow (1) explained above.
     if (!lastUpdateIDFromClient) {
         // If there is a ReconnectApp query in progress, we should not start another one.
-        if (queryPromise) {
+        if (GetMissingOnyxUpdatesPromiseProxy.GetMissingOnyxUpdatesPromise) {
             return;
         }
 
         Log.info('Client has not gotten reliable updates before so reconnecting the app to start the process');
 
         // Since this is a full reconnectApp, we'll not apply the updates we received - those will come in the reconnect app request.
-        queryPromise = App.finalReconnectAppAfterActivatingReliableUpdates();
+        GetMissingOnyxUpdatesPromiseProxy.GetMissingOnyxUpdatesPromise = App.finalReconnectAppAfterActivatingReliableUpdates();
     } else {
         // The flow below is setting the promise to a getMissingOnyxUpdates to address flow (2) explained above.
 
         // Get the number of deferred updates before adding the new one
-        const existingDeferredUpdatesCount = Object.keys(deferredUpdatesProxy.deferredUpdates).length;
+        const existingDeferredUpdatesCount = Object.keys(DeferredUpdates.deferredUpdates).length;
```

Comment:
Can you just add a `DeferredUpdates.isEmpty()` method instead of accessing the dictionary here directly?